crates/iceberg/src/transaction.rs

```diff
@@ -261,7 +286,7 @@ impl<'a> FastAppendAction<'a> {
             let mut manifest_stream = self
                 .snapshot_produce_action
                 .tx
-                .table
+                .current_table
```

Comment:
I realized that there are some functions based on the table. And storing only current metadata may cause inconsistencies easily in the future. So change to store current table instead of metadata